### ISSUE TITLE
Bug: Fix type loading errors in `@eslint/js`

### ISSUE THREAD

**Issue by aryaemami59** [eslint/eslint#19110]
### Environment

Node version: 22.11.0
npm version: 10.9.0
Local ESLint version: 9.14.0
Global ESLint version: N/A
Operating System: Windows 10

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
export default []
```
</details>

In an ESM file I tried to import `configs` from `@eslint/js` which according to TypeScript should work but I get an error.

```ts
import { configs } from '@eslint/js'
```

### What did you expect to happen?

The named imports to resolve properly in ESM files.

### What actually happened?

I received the following `SyntaxError` error:

```bash
SyntaxError: Named export 'configs' not found. The requested module '@eslint/js' is a CommonJS module, which may not support all module.exports as named exports.
```

### Link to Minimal Reproducible Example

https://github.com/aryaemami59/repros/tree/eslint-js-newly-ported-type-definitions-issue

You can clone it by running the following in your CLI:

```bash
git clone -b eslint-js-newly-ported-type-definitions-issue https://github.com/aryaemami59/repros
```

### Participation

- [x] I am willing to submit a pull request for this issue.

### Mismatch in Named Exports with TypeScript's [**`Node16`**](https://www.typescriptlang.org/docs/handbook/modules/reference.html#node16-nodenext-1) ESM Module Resolution

**Problem**: There is a mismatch between the named exports in `@eslint/js` during runtime and the type definitions under TypeScript's [**`Node16`**](https://www.typescriptlang.org/docs/handbook/modules/reference.html#node16-nodenext-1) ESM module resolution strategy. The type definitions indicate the following named exports:

```ts
{
  configs, meta
}
```

But if we try to import these named exports in an ESM file, we learn that they are not available.

**Solution**: We can do change the exports in `src/index.js` from:

```js
module.exports = {
  meta: {
    name: '@eslint/js',
    version,
  },
  configs: {
    all: require('./configs/eslint-all'),
    recommended: require('./configs/eslint-recommended'),
  },
}
```

to:

```js
const all = require('./configs/eslint-all')
const recommended = require('./configs/eslint-recommended')

const configs = {
  all,
  recommended,
}

const meta = {
  name: '@eslint/js',
  version,
}

module.exports = {
  meta,
  configs,
}
```

<details>

  <summary>
  View Diff
  </summary>

```diff
 /**
  * @fileoverview Main package entrypoint.
  * @author Nicholas C. Zakas
  */

 "use strict";

 const { version } = require("../package.json");

 //------------------------------------------------------------------------------
 // Public Interface
 //------------------------------------------------------------------------------

+const all = require("./configs/eslint-all");
+const recommended = require("./configs/eslint-recommended");
+
+const configs = {
+    all,
+    recommended
+};
+
+const meta = {
+    name: "@eslint/js",
+    version
+};
+
 module.exports = {
-    meta: {
-        name: "@eslint/js",
-        version
-    },
-    configs: {
-        all: require("./configs/eslint-all"),
-        recommended: require("./configs/eslint-recommended")
-    }
+    meta,
+    configs
 };
```

</details>

<details>

  <summary><b>Note</b></summary>

You can test for these by following the [Reproduction Steps](#reproduction-steps) running `yarn test`. This also becomes apparent if you run [**Are The Types Wrong**](https://github.com/arethetypeswrong/arethetypeswrong.github.io):

```bash
npx @arethetypeswrong/cli@latest --from-npm @eslint/js
```

Here is a before and after:

#### **Before:**

![Image](https://github.com/user-attachments/assets/2f9bc57e-636b-4a22-8c28-aba8a6300a97)

#### **After:**

![Image](https://github.com/user-attachments/assets/18403b35-4edf-4096-aff9-0315e06664d8)

</details>

## Reproduction Steps

1. Clone this repository.
2. Run `yarn install`.
3. Run `yarn test`.

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
export default []
```
</details>

```js
import { configs } from '@eslint/js'
```

### What did you expect to happen?

The named imports (**`configs`** and **`meta`**) to resolve properly in ESM files.

### What actually happened?

I received the following `SyntaxError` error:

```bash
SyntaxError: Named export 'configs' not found. The requested module '@eslint/js' is a CommonJS module, which may not support all module.exports as named exports.
```

### Link to Minimal Reproducible Example

https://github.com/aryaemami59/repros/tree/eslint-js-newly-ported-type-definitions-issue

### Participation

- [x] I am willing to submit a pull request for this issue.

### Additional comments

This is similar to #18920, I have the fix for this already I will submit a PR shortly.

**Comment by mdjermanovic** [comment]
> import { configs } from '@eslint/js'
> 
>     
>   
> 
> ### What did you expect to happen?
> 
> The named imports (**`configs`** and **`meta`**) to resolve properly in ESM files.

This isn't intended to work. As described in the docs, `@eslint/js` should be used this way:

```js
import js from "@eslint/js";

// ...use js.configs
```

> **Problem**: There is a mismatch between the named exports in `@eslint/js` during runtime and the type definitions under TypeScript's [**`Node16`**](https://www.typescriptlang.org/docs/handbook/modules/reference.html#node16-nodenext-1) ESM module resolution strategy. The type definitions indicate the following named exports:
> 
> {
>   configs, meta
> }

Can this be fixed in types?

**Comment by aryaemami59** [comment]
@mdjermanovic

> Can this be fixed in types?

I don't believe so and even if we could, I think simply allowing to use named imports seems like a much more reasonable solution.

**Comment by mdjermanovic** [comment]
I think no updated are needed here. @eslint/eslint-team thoughts?

**Comment by nzakas** [comment]
I agree with @mdjermanovic. This isn't how the package is intended to be used.

**Comment by aryaemami59** [comment]
@nzakas That's fair, can I at least try to fix the type definitions so they match the intended runtime behavior?
